prime-router/metadata/HL7/catchall/hl7/segments/MSH/MessageHeader.yml

```diff
@@ -3,9 +3,9 @@
 resourceType: MessageHeader
 
 id:
-  type: NAMED_UUID
-  valueOf: MSH.10
-  expressionType: HL7Spec
+  type: STRING
+  valueOf: "GeneralUtils.generateResourceId()"
+  expressionType: JEXL
```

Comment:
Isn't MSH-10 the one we use to trace things back to their origin and stuff?

Reply:
Looks like it's beeping FHIR->HL7 mapped here: https://github.com/CDCgov/prime-reportstream/blob/81020792ccd8cba1027eaafc3b3ee99905d3b4af/prime-router/src/main/resources/metadata/hl7_mapping/resources/MessageHeader/MSH.yml#L46-L49 Should I add it to mshextension just for consistency/fallback?

Reply:
I meant that either we should depend on `Bundle.identifier` entirely for MSH.10 and not include MSH.10 in MSHExtension or we should keep MSH.10 in MSHExtension and also use it in the FHIR to HL7 then delete the use of `Bundle.identifier`.

I lean towards using `Bundle.identifier` and not including MSH.10 in MSHExtension.

Reply:
Ah I see. Yeah I would agree that we should remove it from MSHExtension in that case. Is there any other discussion that should happen before I yoink it out?

Reply:
Cool cool, I think just add a comment at the top of MessageHeader about the use of Bundle.Identifier and we should be good

Reply:
Reverted the MSHExtension changes for MSH.10: https://github.com/CDCgov/prime-reportstream/pull/15333/commits/a2138d7cbc77e75e2cf3d6abab939c674c16ac6f
Apparently there was already a note in MessageHeader, but I moved it up: https://github.com/CDCgov/prime-reportstream/pull/15333/commits/2c2453ccd1dd6b1c035661f95e9b1f8367c06e19
To do: Updating a few of the other tests to remove the MSH.10 extension.

Reply:
Removed MSH.10 from the rest of the test files: https://github.com/CDCgov/prime-reportstream/pull/15333/commits/0c3e02af5f445b0c46fa6a095583e52d71d50a64

Reply:
Just had an alarming thought while going over final checks for this PR. If this change to explicitly generate an id here were to cause any issues, it wouldn't be caught until staging or production because our tests essentially ignore fhir resource ids as a value. Not sure I'm able to think this through at the moment, but would be interested to hear thoughts @mkalish @brick-green

Reply:
While looking over everything one last time before requesting more eyes on the changes, I looked at the reasoning behind MSH.10 again. Based on the wording of the comment for MessageHeader we may need this after all, and some bonus logic to make it work: 

> "Need a gForge to request a MessageHeader.identifier (https://gforge.hl7.org/gf/project/fhir/tracker/?action=TrackerItemEdit&tracker_item_id=20704&start=0)
Note that if a message is part of a batch, then the MSH goes to a ""sub-bundle""..
Note that as messages switch from V2 to FHIR and the response back from FHIR to V2, the ""mapping engine"" needs to track the actual MSH.10.00.00 value and Bundle.identifier value to connect responses correctly."

https://docs.google.com/spreadsheets/d/13pgda5xl-PwCgB9j0axyymwwv7RJVcrIzY8Ah1y1Y1M/edit?gid=0#gid=0&range=S16

Reply:
Brought this up as a 17th in standup. It was concluded that I was worried for nothing.
With the first question about altering the mapping for MessageHeader.id, if this caused any problems it would be caught in any HL7->FHIR->HL7 mappings tests. 
For the note on MSH.10 in the mapping inventory, this case does not currently apply to ReportStream so it is a non-issue.